src/actions/albums.js

```diff
@@ -45,6 +46,46 @@ export const fetchAlbums = (mangoIndex) => {
   }
 }
 
+// Returns albums photos informations from the album ID
+export const fetchAlbumPhotosStatsById = (albumId) => {
```

Comment:
Oh, after looking at the code I'm afraid we didn't understand each other. [Here's essentially what I had in mind](https://gist.github.com/y-lohse/f44b9c6e9eaffff28bd1ca8ef9f57766) (warning: untested).

If I understood the code correctly, the two versions should work exactly the same, but the version without promises avoids duplicating the catch/throw part, and reads more "naturally".